src/ja/user/config/query.py

```diff
@@ -12,70 +13,297 @@ class QueryCommandConfig(UserConfig):
     matches any of the provided values.
     """
 
+    def __init__(self, uid: List[str] = None, label: List[str] = None, owner: List[str] = None,
+                 priority: List[JobPriority] = None, status: List[JobStatus] = None, is_preemtible: bool = None,
+                 special_resources: List[List[str]] = None, cpu_threads: Tuple[int, int] = (1, 1),
+                 memory: Tuple[int, int] = (1, 1), before: datetime = None, after: datetime = None,
+                 ssh_config: SSHConfig = None, verbosity: Verbosity = Verbosity.NONE):
+        if cpu_threads[0] > cpu_threads[1] or cpu_threads[0] < 1:
+            raise ValueError("Invalid range for cpu_threads.")
+        if memory[0] > memory[1] or memory[0] < 1:
+            raise ValueError("Invalid range for memory.")
+        if after is not None and before is not None:
+            if after < before:
+                raise ValueError("incorrect datetime")
```

Comment:
Your error messages are not very helpful to be honest.
Please write more detailed error messages that give more insight into why the parameters are not valid.